### PR TITLE
[@types/lodash] Fix _.get backwards compatibility

### PR DESCRIPTION
**Pull request by atsikov**
A few fixes to `_.get` types:
* Return `any` when a generic `string` path is provided instead of a literal for backwards campatibility
* Return correct non-nullable type when default value is provided
* Copy types from `@types/lodash@4.14.186` as TS < 4.2 compatible

---

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: a follow-up to a previously merged `_.get` types improvements https://github.com/DefinitelyTyped/DefinitelyTyped/pull/62968

### CODE REVIEW

File: types/lodash/lodash-tests.ts
Comment (lizheming) [inline types/lodash/lodash-tests.ts:5315]: May I ask why we need `as string`?

File: types/lodash/lodash-tests.ts
Comment (lizheming) [inline types/lodash/lodash-tests.ts:5351]: Is it possible to definition return type only `PrimitiveChain<false>`?

File: types/lodash/lodash-tests.ts
Comment (lizheming) [reply types/lodash/lodash-tests.ts:5315]: OK, I know it. `as string` then the key type can't point to `a` string.

File: types/lodash/lodash-tests.ts
Comment (atsikov) [reply types/lodash/lodash-tests.ts:5351]: `{ a: false }` is inferred as `{ a: boolean }` in this case.
In case of stricter `_.chain({ a: false } as const).get("a");`, it would return only `PrimitiveChain<false>`

Comment (lizheming) [reply types/lodash/lodash-tests.ts:5351]: Got it! thank you for your answer😘

File: types/lodash/common/object.d.ts
Comment (aj-r) [inline types/lodash/common/object.d.ts:1135]: Fix indentation

**Review by aj-r — APPROVED** [review]

**Comment by aj-r** [comment]
> Am I right, `v3` should not be added to `ts4.1` as it is used as a separate module?

I believe you are correct

**Comment by atsikov** [comment]
@aj-r `typescript-bot` says "[lodash/ts4.1/OTHER_FILES.txt](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63025/files/57c00720d80097223d9e80ac794b01f0d077b83d#diff-115d139631957667f487249953d1e23d144841585ca593b6b0ff30719d18014d): not [the expected form](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-other_filestxt)"
Do you know what might be an issue? Eg, should it contain file paths relative to the package root, and not to the file itself?
Or should `ts4.1/<file>.d.ts` entries be added to the root OTHER_FILES instead of this one?

**Comment by aj-r** [comment]
> @aj-r `typescript-bot` says "[lodash/ts4.1/OTHER_FILES.txt](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63025/files/57c00720d80097223d9e80ac794b01f0d077b83d#diff-115d139631957667f487249953d1e23d144841585ca593b6b0ff30719d18014d): not [the expected form](https://github.com/DefinitelyTyped/DefinitelyTyped#user-content-other_filestxt)" Do you know what might be an issue? Eg, should it contain file paths relative to the package root, and not to the file itself? Or should `ts4.1/<file>.d.ts` entries be added to the root OTHER_FILES instead of this one?

No idea, we might need help from a DefinitelyTyped maintainer.

**Review by sandersn — APPROVED** [review]
This change is fine, with one caveat and one explanation:

1. The bot complains whenever OTHER_FILES.txt is added, since those files are not guaranteed to be compiled and tested. It doesn't cause tests to fail, but it requires signoff from a DT admin. But changing lodash at all requires that.
2. DT will stop supporting Typescript 4.1 when 4.9 ships in a week or so. This PR would be significantly smaller without the 4.1 copy, and `@types/lodash` would be significantly smaller. So it might be worth waiting.

**Comment by atsikov** [comment]
Thanks for the explanation @sandersn

The thing is, the change itself is compatible with TS 4.1, and technically doesn't require "typesVersion" at all. However the previous change raised compatibility to ">=4.1" because of template literals, so this is an attempt to make the package backwards compatible for people that for some reason still use "<4.1", as generally such compatibility shouldn't change in a patch version.

**Comment by atsikov** [comment]
Ready to merge

**Pull request merged**
